File: javascript/1415.js
/**
 * @param {number} n
 * @param {number} k
 * @return {string}
 */
var getHappyString = function (n, k) {
    const alpha = "0abc"
    var find = function (res, i, n) {
        if (i == n) {
            k--;
            return k == 0;
        }
        for (let c = 1; c < 4; c++)
            if (res[i - 1] != alpha[c]) {
                res[i] = alpha[c];
                if (find(res, i + 1, n))
                    return true;
            }
        return false;
    };
    let res = new Array(n);
    for (let c = 1; c < 4; c++) {
        res[0] = alpha[c];
        if (find(res, 1, n))
            return res.join("");
    }
    return "";
};

File: c/1415.c
int k;

bool find(char* res, int i, int n) {
    if (i == n) {
        k--;
        return k == 0;
    }
    for (char c = 'a'; c < 'd'; c++)
        if (res[i - 1] != c) {
            res[i] = c;
            if (find(res, i + 1, n))
                return true;
        }
    return false;
}

char* getHappyString(int n, int kk) {
    char* res = (char*)malloc(sizeof(char) * (n + 1));
    res[n] = '\0';
    k = kk;
    for (char c = 'a'; c < 'd'; c++) {
        res[0] = c;
        if (find(res, 1, n))
            return res;
    }
    res[0] = '\0';
    return res;
}
